pkg/gadget-collection/gadgets/top/block-io/gadget.go

```diff
@@ -103,35 +107,46 @@ func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
 
 	traceName := gadgets.TraceName(trace.ObjectMeta.Namespace, trace.ObjectMeta.Name)
 
-	maxRows := types.MaxRowsDefault
-	intervalSeconds := types.IntervalDefault
+	maxRows := top.MaxRowsDefault
+	intervalSeconds := top.IntervalDefault
 	sortBy := types.SortByDefault
 
 	if trace.Spec.Parameters != nil {
 		params := trace.Spec.Parameters
 		var err error
 
-		if val, ok := params[types.MaxRowsParam]; ok {
+		if val, ok := params[top.MaxRowsParam]; ok {
 			maxRows, err = strconv.Atoi(val)
 			if err != nil {
-				trace.Status.OperationError = fmt.Sprintf("%q is not valid for %q", val, types.MaxRowsParam)
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for %q", val, top.MaxRowsParam)
 				return
 			}
 		}
 
-		if val, ok := params[types.IntervalParam]; ok {
+		if val, ok := params[top.IntervalParam]; ok {
 			intervalSeconds, err = strconv.Atoi(val)
 			if err != nil {
-				trace.Status.OperationError = fmt.Sprintf("%q is not valid for %q", val, types.IntervalParam)
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for %q", val, top.IntervalParam)
 				return
 			}
 		}
 
-		if val, ok := params[types.SortByParam]; ok {
-			sortBy, err = types.ParseSortBy(val)
-			if err != nil {
-				trace.Status.OperationError = fmt.Sprintf("%q is not valid for %q", val, types.SortByParam)
-				return
+		if val, ok := params[top.SortByParam]; ok {
+			sortByColumns := strings.Split(params[top.SortByParam], ",")
+			sortBy = make([]string, len(sortByColumns))
+
+			cols := columns.MustCreateColumns[types.Stats]()
+			for i, col := range sortByColumns {
+				colToTest := col
+				if len(col) > 0 && col[0] == '-' {
+					colToTest = colToTest[1:]
+				}
+				_, ok := cols.GetColumn(colToTest)
+				if !ok {
+					trace.Status.OperationError = fmt.Sprintf("%q is not valid for %q", val, top.SortByParam)
+					return
+				}
+				sortBy[i] = col
```

Comment:
Well it looks like that because it is the same :D
Was the first version of this check before i found `VerifyColumnNames`. Forgot to update it in the `pkg/gadget-collection` 

Will change